tests/suite/test_v_s_route_redirects.py

```diff
@@ -0,0 +1,84 @@
+from suite.custom_resources_utils import get_vs_nginx_template_conf, patch_v_s_route_from_yaml
+from suite.resources_utils import get_first_pod_name, get_events, wait_before_test
+
+
+@pytest.mark.vsr
+@pytest.mark.parametrize('crd_ingress_controller, v_s_route_setup',
+                         [({"type": "complete", "extra_args": [f"-enable-custom-resources"]},
+                           {"example": "virtual-server-route-redirects"})],
+                         indirect=True)
+class TestVSRRedirects:
+    def test_config(self, kube_apis, ingress_controller_prerequisites, crd_ingress_controller, v_s_route_setup):
+        wait_before_test(1)
+        ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+        config = get_vs_nginx_template_conf(kube_apis.v1,
+                                            v_s_route_setup.namespace,
+                                            v_s_route_setup.vs_name,
+                                            ic_pod_name,
+                                            ingress_controller_prerequisites.namespace)
+        assert 'return 307 ' in config and 'return 301 ' in config
+
+    def test_custom_canned_response(self, kube_apis, crd_ingress_controller, v_s_route_setup):
+        req_host = f"{v_s_route_setup.public_endpoint.public_ip}:{v_s_route_setup.public_endpoint.port}"
+        req_url = f"http://{req_host}{v_s_route_setup.route_m.paths[0]}?arg1=arg"
+        wait_and_assert_status_code(307, req_url, v_s_route_setup.vs_host, allow_redirects=False)
+        resp = requests.get(req_url, headers={"host": v_s_route_setup.vs_host}, allow_redirects=False)
+        assert resp.headers['location'] == "http://example.com"
+
+    def test_default_canned_response(self, kube_apis, crd_ingress_controller, v_s_route_setup):
+        req_host = f"{v_s_route_setup.public_endpoint.public_ip}:{v_s_route_setup.public_endpoint.port}"
+        req_url = f"http://{req_host}{v_s_route_setup.route_m.paths[1]}"
+        wait_and_assert_status_code(301, req_url, v_s_route_setup.vs_host, allow_redirects=False)
+        resp = requests.get(req_url, headers={"host": v_s_route_setup.vs_host}, allow_redirects=False)
+        assert resp.headers['location'] == f"http://{v_s_route_setup.vs_host}/backends/default-redirect?arg="
+
+    def test_update(self, kube_apis, crd_ingress_controller, v_s_route_setup):
+        req_host = f"{v_s_route_setup.public_endpoint.public_ip}:{v_s_route_setup.public_endpoint.port}"
+        req_url_1 = f"http://{req_host}{v_s_route_setup.route_m.paths[0]}"
+        req_url_2 = f"http://{req_host}{v_s_route_setup.route_m.paths[1]}"
+        wait_before_test(1)
+        vs_name = f"{v_s_route_setup.namespace}/{v_s_route_setup.vs_name}"
+        vsr_name = f"{v_s_route_setup.namespace}/{v_s_route_setup.route_m.name}"
+        vsr_event_text = f"Configuration for {vsr_name} was added or updated"
+        vs_event_text = f"Configuration for {vs_name} was added or updated"
```

Comment:
Tiny optional suggestion:
I think it makes sense to swap the order of these lines.